ios-target-app/Home/Views/HomeViewController.swift

```diff
@@ -6,13 +6,107 @@
 //
 
 import UIKit
+import MapKit
+import Combine
 
-class HomeViewController: UIViewController {
+class HomeViewController: UIViewController, MKMapViewDelegate {
+  
+  private lazy var mapView: MKMapView = {
+    let map = MKMapView()
+    map.overrideUserInterfaceStyle = .dark
+    map.isZoomEnabled = true
+    map.isScrollEnabled = true
+    return map
+  }()
+  
+  private let locationManager: LocationManager
+  private var subscribers = Set<AnyCancellable>()
+  
+  init(locationManager: LocationManager) {
```

Comment:
Have you considered using a different design approach here? Maybe MVVM or similar

Reply:
You mean like addressing this inside a ViewModel instead?